spec/tbd_tests_spec.rb

```diff
@@ -1413,8 +1419,7 @@
           sub2.each do |sb2|
             # With default tolerances, none of the subsurface edges "match" up.
             expect(TBD.matches?(sb1, sb2)).to be(false)
-
-            # Greater tolerances trigger 5x matches, as follows:
+            # Greater tolerances however trigger 5x matches, as follows:
             # "Sub Surface 7" ~ "Sub Surface 8" ~ "Sub Surface 6"
             # "Sub Surface 3" ~ "Sub Surface 5" ~ "Sub Surface 4"
             # "Sub Surface 1" ~ "Sub Surface 2"
```

Comment:
In the 'seb.osm' model:

- "Sub Surface 5" is sandwiched between "Sub Surface 3" and "Sub Surface 4"
- "Sub Surface 8" is sandwiched between "Sub Surface 7" and "Sub Surface 6"

With a user-set tolerance of 100mm, both window jambs of "Sub Surface 5" and "Sub Surface 8" would be reset as (mild) transitions (PSI factor of 0.000 W/K per m). The matching (nearby) jambs of the other subsurfaces would also be reset as (mild) transitions.

Reply:
Is there anything in Topolys that could help with this?  Like allowing a separate tolerance for subsurfaces?  Should this also apply to windows that are close to a corner (and so would share common framing)?

Reply:
@macumber : Good suggestion. Given the (very) basic, constrained [rules](https://github.com/rd2/tbd/blob/6ee3d7b106f8e6f8d89df7097549baab61be2572/lib/tbd/psi.rb#L1772) under which the proposed solution currently operates, I think we're fine. If the solution were to extend to 2x _nearby_, parallel subsurface edges yet of different length (e.g. 1x door jamb in close proximity to 1x _shorter_ side window jamb), then I would see a need for Topolys to insert additional vertices along the door rough opening. Similarly, for 2x strip window jambs _nearly_ connected along a wall corner, it would be nice to have Topolys insert 2x additional vertices along the wall corner edge itself, aligned with the window jamb vertices (allowing TBD to ignore the _nearby_ shared wall edge length as a _corner_ thermal bridge). If the _nearly_ adjoined strip windows are not of the same height, then 3 or 4 vertices may need to be inserted along the wall corner edge. So this can easily snowball into something more complicated. It becomes even more challenging when the surfaces are tilted!

We shouldn't lose sight of why this feature is being added in the first place. The Figure illustrates a real case where a 3rd-party application ends up automatically splitting a single window into 2x smaller windows (as a result of automatically splitting a concave wall surface into 2x convex wall surfaces). This is a questionable outcome, but it happens. If the 2x windows actually shared vertices along their common edge, TBD would (correctly) consider the 2x subsurfaces as 1x. Without the new proximity tolerance, TBD instead treats both subsurfaces as unique and would process 4x window jambs (exaggerating heat losses from thermal bridging). The new feature allows TBD to treat the 2x split subsurfaces as one (2x window jambs instead of 4x).
<img width="567" alt="ies" src="https://user-images.githubusercontent.com/98104985/210446877-40083be0-b655-41ef-aee2-ddafe2098888.png">

Long story short, the feature is there to accommodate odd (yet unfortunately common) cases like this one. In other circumstances, it provides some tolerance for legacy OpenStudio models. I'm a tad weary of overextending the feature - it shouldn't become a crutch for poor modelling techniques or questionable (automated) processes by 3rd-parties. It's up to you - I'll gladly incorporate/test any new Topolys features you may want to suggest.

Reply:
Thanks Denis, I think the option to split longer edges when there are nearby sub surfaces might be generally useful it doesn't seem like it would "solve" this case.  With as many crazy cases as these, it really seems like a way for the modeler to visualize the TDB JSON and make tweaks to it would be more useful than trying to automatically get everything right.

Reply:
I can't see fully automating all the heat loss calculations without something near full CAD drawings including all the studs, window frames, etc.

Reply:
I opened an issue in Topolys for adding vertices on edges for nearby holes.

https://github.com/automaticmagic/topolys/issues/23

Reply:
- Correct: splitting a longer opaque surface edge (near 2x subsurface jamb edges) only makes sense if it's a corner. In the illustrated example, the taller surface edge splitting the 2x windows would be tagged as a (mild) _transition_ in any case (i.e. PSI 0 W/K per m) - there would be no point in splitting the taller edge into 3x sections.
- True, having users right-click/edit an edge (e.g. via SU plugin), which would require interacting with TBD/Topolys, would be great. In fact, it's an old(er) idea that's been in circulation for a while. Without that in place, users can still script their way in editing a TBD JSON object/file. Nonetheless, it remains fair to try to automate default behaviours whenever.